tests/bson/bson-decimal128_error-001.phpt

```diff
@@ -11,7 +11,7 @@ require_once __DIR__ . '/../utils/basic.inc';
 
 echo throws(function() {
     new MongoDB\BSON\Decimal128([]);
-}, MongoDB\Driver\Exception\InvalidArgumentException::class), "\n";
+}, TypeError::class), "\n";
```

Comment:
I'm not entirely sure about these changes - some people may see this as a BC break. I'll have to double-check why the exception class changes for this and ObjectId.

Reply:
Does this change apply to _all_ PHP versions? It calls to mind some of the variations between PHP 7.x and 8.0 when we were implementing [PHPC-1631](https://jira.mongodb.org/browse/PHPC-1631).

Reply:
It affects all PHP versions, since we unconditionally add argument types to methods. I'll take another look at this tomorrow to figure out where this is coming from.

Reply:
I'm pretty sure this is the result of parameter types now being validated by arginfo instead of zpp. That's probably something we should document in release notes (you can add `documentation-needed` on the ticket and make a note of this in a comment). I don't think there's any place to mention this in the extension docs, though -- it'd be pedantic to add this to _every_ method's changelog.

Reply:
Yes, I believe adding this to each method's changelog would be overkill. Maybe a paragraph in the release notes would be appropriate for this.